ompi/mpi/fortran/base/gen-mpi-sizeof.pl

```diff
@@ -154,15 +156,21 @@ sub generate {
     queue_sub("integer(int${size})", "int${size}", "int${size}");
 }
 for my $size (qw/16 32 64 128/) {
-    if (!($size == 16 && $mpi_real2 == 0) &&
-        !($size == 128 &&$mpi_real16 == 0)) {
+    if (!($size == 16 && $mpi_iso_real16 == 0) &&
+        !($size == 128 && $mpi_real16 == 0)) {
         queue_sub("real(real${size})", "real${size}", "real${size}");
     }
-    if (!($size == 16 && $mpi_complex4 == 0) &&
+    if (!($size == 16 && $mpi_iso_real16 == 0) &&
         !($size == 128 && $mpi_complex32 == 0)) {
         queue_sub("complex(real${size})", "complex${size}", "real${size}");
     }
 }
+if ($mpi_real2 == 1) {
+    queue_sub("real*2", "real2");
+}
+if ($mpi_complex4 == 1) {
+    queue_sub("complex*4", "complex4");
+}
```

Comment:
When a Fortran compiler supports both `REAL16` and `REAL*2`, both `MPI_Sizeof_real16_*` and `MPI_Sizeof_real2_*` are defined by this code. Fujitsu Fortran compiler supports both and outputs warning `Specific procedures (MPI_Sizeof_real16_scalar) and (MPI_Sizeof_real2_scalar) do not ensure that generic reference (MPI_Sizeof) is unambiguous.` when an MPI program includes `mpif.h`. Defining two functions itself is not a problem, but having the same signature for the `MPI_Sizeof` *interface* is the problem.

Also, the subroutine name `MPI_Sizeof_real16_*` represents bits and `MPI_Sizeof_real2_*` represents bytes is confusing.

How about:

```perl
if ($mpi_real2 == 1 && $mpi_iso_real16 == 0) {
    queue_sub("real*2", "real16");
}
if ($mpi_complex4 == 1 && $mpi_iso_real16 == 0) {
    queue_sub("complex*4", "complex16");
}
```